azurerm/data_source_network_interface.go

```diff
@@ -0,0 +1,273 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetworkInterfaceRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
since this is a data source (and can't be specified), we can just make this `Computed: true`

Reply:
Make sense, done.